### PR TITLE
[DYN-3579] Preferences General tab functionality

### PR DESCRIPTION
**Pull request by Astul-Betizagasti**
### Purpose

Add functionality to the controls on the Preferences window's General tab

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )

### FYIs

Roberto Tellez (@RobertGlobant20 )

### CODE REVIEW

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 1]
```diff
@@ -382,7 +382,10 @@ public string DefaultPythonEngine
         /// If enabled user's custom package locations will not be loaded.
         /// </summary>
         public bool DisableCustomPackageLocations { get; set; }
-
+        /// <summary>
+        /// If true the run mode will be set to Automatically, if false to Manual.
+        /// </summary>
+        public bool RunTypeAutomatic { get; set; }
```
Comment (QilongTang) [inline src/DynamoCore/Configuration/PreferenceSettings.cs:388]: Can we serialize this info using string so we can just borrow `enum RunType { Manual, Automatic, Periodic }`? What would be the trade from your point of view?

File: src/DynamoCore/Models/DynamoModel.cs [hunk 2]
```diff
@@ -2591,6 +2593,12 @@ public void ClearCurrentWorkspace()
             OnWorkspaceClearing();
 
             CurrentWorkspace.Clear();
+            if (CurrentWorkspace is HomeWorkspaceModel)
+            {
+                _ = PreferenceSettings.RunTypeAutomatic
```
Comment (QilongTang) [inline src/DynamoCore/Models/DynamoModel.cs:2598]: can you add comments for this block?

File: src/DynamoCoreWpf/UI/Converters.cs [hunk 3]
```diff
@@ -1840,18 +1840,59 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
...
+                case "f1":
+                    return Resources.DynamoViewSettingMenuNumber00;
+                case "f2":
+                    return Resources.DynamoViewSettingMenuNumber000;
+                case "f3":
+                    return Resources.DynamoViewSettingMenuNumber0000;
+                case "f4":
+                    return Resources.DynamoViewSettingMenuNumber00000;
+                default:
+                    return null;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/Converters.cs:1875]: I feel we should also have a bunch of legacy code we can obsolete/ delete here? The old number format comes with a dialog, let's delete it?

File: src/DynamoCore/Models/DynamoModel.cs (hunk 2 above)
Comment (Astul-Betizagasti) [reply src/DynamoCore/Models/DynamoModel.cs:2598]: sure, will do on next update

File: src/DynamoCore/Models/RunSettings.cs [hunk 4]
```diff
@@ -106,7 +106,6 @@ public RunSettings(RunType runType, int period)
         internal void Reset()
         {
             RunEnabled = true;
-            RunType = RunType.Automatic;
```
[the remark this answers is not in the document]
Comment (Astul-Betizagasti) [reply src/DynamoCore/Models/RunSettings.cs:109]: this function is called every time a "new" home workspace is open (actually it is the same workspace that gets reset every time) , and it was overriding the user's default run setting.

File: src/DynamoCoreWpf/UI/Converters.cs (hunk 3 above)
Comment (Astul-Betizagasti) [reply src/DynamoCoreWpf/UI/Converters.cs:1875]: Hello Aaron, can you explain further? Previously the node format was chosen from a sub-menu in the Settings Dynamo menu which we removed already. This preference is saved on the PreferencesSettings with the codes "f1", "f2", etc but shown to the user as 0.0, 0.00, etc, so this converter is used to convert back and forth between the two formats.

File: src/DynamoCore/Configuration/PreferenceSettings.cs (hunk 1 above)
Comment (Astul-Betizagasti) [reply src/DynamoCore/Configuration/PreferenceSettings.cs:388]: This variable is the one that saves the user's Default run setting, and being that the only options the user have are Manual or Automatic it doesn't feel right having the option to save Periodic on it too. On the other hand I believe using the enum as you suggested would make it more manageable in terms of code. I suppose the best option here depends on what we value the most, what I can say for sure is that the name of the property should be more representative of what it saves so I will change it on the next update to something more fitting

Comment (QilongTang) [reply src/DynamoCore/Configuration/PreferenceSettings.cs:388]: @Astul-Betizagasti we can use `IsDefaultRuntypeAuto` etc, but that still looks less intuitive to me and less expandable. I prefer we serialize the actual type, even it is only going to either `Automatic` or `Manual` for now, but this allows us to expand in the future, e.g. when Dynamo become a web service, user could have time based run type etc.

File: src/DynamoCoreWpf/UI/Converters.cs (hunk 3 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/UI/Converters.cs:1875]: ah OK, my bad then that I did not notice we are already re-using the resources

File: src/DynamoCore/Configuration/PreferenceSettings.cs (hunk 1 above)
Comment (Astul-Betizagasti) [reply src/DynamoCore/Configuration/PreferenceSettings.cs:388]: Make sense to me, I will update that

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 5]
```diff
@@ -382,7 +383,10 @@ public string DefaultPythonEngine
         /// If enabled user's custom package locations will not be loaded.
         /// </summary>
         public bool DisableCustomPackageLocations { get; set; }
-
+        /// <summary>
+        /// If true the run mode will be set to Automatically, if false to Manual.
```
Comment (QilongTang) [inline src/DynamoCore/Configuration/PreferenceSettings.cs:387]: Please update the summary as well

Comment (Astul-Betizagasti) [reply src/DynamoCore/Configuration/PreferenceSettings.cs]: good catch, updated!

**Review by QilongTang — APPROVED** [review]
LGTM

**Comment by QilongTang** [comment]
@Astul-Betizagasti Do you mind adding unit test for `RunSetting`, you can search for tests for preference setting which cover serialization and deserialization.

**Comment by Astul-Betizagasti** [comment]
@QilongTang, I just made a commit addressing the changes to the tests as you asked. The new property was added as part of the checks made by the existing test related to serializing the preferences settings

**Comment by QilongTang** [comment]
@Astul-Betizagasti Thanks!

**Pull request merged**